test/SILOptimizer/definite_init_diagnostics.swift

```diff
@@ -83,11 +83,6 @@ func test2() {
   }
   b4 = 7
   
-  let b5: Any
```

Comment:
Why did you remove this test case?

Reply:
The change to Sema makes the DI test redundant, and furthermore, since the error here gets diagnosed in Sema, it prevents any other DI tests in this file from being reached.